### PR TITLE
[FIXED] model command help message error

### LINKED ISSUE #9470: [mesheryctl] crash in experimental model command
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior
```
➜  mesheryctl git:(master) ./mesheryctl exp model
View list of models and detailed information of a specific model

Usage:
  mesheryctl exp model [flags]
  mesheryctl exp model [command]

Examples:

// To view list of components
mesheryctl exp model list

// To view a specific model
mesheryctl exp model view [model-name]


Available Commands:
  list        list models
  search      search models
  view        view model

Flags:
  -h, --help   help for model

Global Flags:
      --config string   path to config file (default "/Users/lee/.meshery/config.yaml")
  -v, --verbose         verbose output

Use "mesheryctl exp model [command] --help" for more information about a command.
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/layer5io/meshery/mesheryctl/internal/cli/root/system.glob..func40(0xc00037ae00?, {0x104f04060?, 0x0, 0x0})
	/Users/lee/code/meshery/mesheryctl/internal/cli/root/system/model.go:360 +0x2fd
github.com/spf13/cobra.(*Command).execute(0x104d13500, {0x104f04060, 0x0, 0x0})
	/Users/lee/code/go/bin/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0x104d0a540)
	/Users/lee/code/go/bin/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/lee/code/go/bin/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/layer5io/meshery/mesheryctl/internal/cli/root.Execute()
	/Users/lee/code/meshery/mesheryctl/internal/cli/root/root.go:82 +0x45
main.main()
	/Users/lee/code/meshery/mesheryctl/cmd/mesheryctl/main.go:26 +0x13
```

### Environment

- Host OS: Mac Linux Windows
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v
- Platform: Docker Kubernetes

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by zakisk**
**Notes for Reviewers**

This PR fixes #9470 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/system/model.go [hunk 1]
```diff
@@ -356,6 +356,7 @@ mesheryctl exp model view [model-name]
 	RunE: func(cmd *cobra.Command, args []string) error {
 		if len(args) == 0 {
 			_ = cmd.Help()
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/model.go:358]: return err?

Comment (zakisk) [reply mesheryctl/internal/cli/root/system/model.go:358]: showing help message when no arguments are passed, so `return nil` will be viable.

File: mesheryctl/internal/cli/root/system/model.go [hunk 2]
```diff
@@ -356,6 +356,7 @@ mesheryctl exp model view [model-name]
...
+			return nil
```
Comment (VihasMakwana) [inline mesheryctl/internal/cli/root/system/model.go:359]:
```suggestion
			return cmd.Help()
```

cmd.Help() will return `nil` so we can directly use it like above I guess.

Comment (zakisk) [reply mesheryctl/internal/cli/root/system/model.go:359]: yes it can be

Comment (zakisk) [reply mesheryctl/internal/cli/root/system/model.go]: @leecalcote I have to use `cmd.Help` func to show help message

File: mesheryctl/internal/cli/root/system/model.go [hunk 3]
```diff
@@ -355,7 +355,7 @@ mesheryctl exp model view [model-name]
 	`,
 	RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/system/model.go:356]: I believe, what @leecalcote hinting is to have a PreRunE function, which can do all these checks? or rather an Args function which can check whether the args passed are correct or not.
for e.g. refer to line no: 171 or 279

Comment (VihasMakwana) [reply mesheryctl/internal/cli/root/system/model.go:356]: @hexxdump yeah, this can be another enhancement.

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/model.go:356]: Please do refactor this PR to leverage Cobra's built-in capabilities. Foremost, look to how other `mesheryctl` commands handle this and implement consistently here.

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/model.go:356]: @zakisk make sense?

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/model.go:356]: @zakisk checking in again...

**Comment by abhijeetgauravm** [comment]
Hey @zakisk  Let's discuss this on Meshery Development Meeting today at 8:30 PM IST / 9 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#).

File: mesheryctl/internal/cli/root/system/model.go (hunk 3 above)
Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/model.go:356]: PR abandoned. 

@hexxdump do you think that you can carry this work forward in a separate PR?

**Comment by leecalcote** [comment]
@Althaf66 will you review this? // @RipulHandoo

**Comment by RipulHandoo** [comment]
@leecalcote This issue has been addressed in [PR #10118](https://github.com/meshery/meshery/pull/10118). I recommend closing this PR.

**Comment by leecalcote** [comment]
Thank you, @RipulHandoo

**Pull request closed without merging**